Java EE Web Profile based application.

violetagg opened the pull request:
This is a test application for https://github.com/cloudfoundry-community/tomee-buildpack/issues/5

cgfrost commented:
Hi, this all looks good to me but I'm concerned that it doesn't provide any of the common endpoints that all the other sample applications provide. @nebhale What do you think? 

Common endpoints listed here: https://github.com/cloudfoundry/java-test-applications#output-content

violetagg commented:
Hi,

I added the common endpoint.
I touched a bit the core module so that I can use it in Java EE environment.
I also added support for FAIL_INIT and FAIL_OOM

Regards,
Violeta

violetagg commented:
Hi,

Is there anything else that I can do in addition for this change?

Thanks a lot,
Violeta

nebhale commented on core/src/main/java/com/gopivotal/cloudfoundry/test/core/MemoryUtils.java:
```diff
@@ -35,7 +35,7 @@
  */
 @Component
-public final class MemoryUtils {
+public class MemoryUtils {
```
Is the a CDI thing?  That the type can't be `final`?

nebhale commented on core/src/main/java/com/gopivotal/cloudfoundry/test/core/MemoryUtils.java:
```diff
@@ -63,13 +63,17 @@ public MemoryUtils() {
      *
      * @return Never returns as it will generate an {@link OutOfMemoryError}
      */
-    @PostConstruct
```
And that it has to return `void`?

violetagg commented:
Yes that's CDI [1] and it uses annotations defined in JSR 250: Common Annotations for the Java platform. According to this specification [2]:

"
2.5 javax.annotation.PostConstruct
...
- The return type of the method MUST be void
- The method MAY be final or non-final, except in the case of EJBs where it MUST be
non-final.
"

Regards,
Violeta

[1] http://docs.oracle.com/javaee/6/tutorial/doc/gmgkd.html
[2] http://download.oracle.com/otn-pub/jcp/common_annotations-1_2-mrel2-spec/jsr-250-1.2-final.pdf?AuthParam=1456943655_a49f931f0ce762f5042d91f31927484b
